blog/_posts/2017-02-25-scalafix-v0.3.md

```diff
@@ -0,0 +1,155 @@
+---
+layout: blog
+post-type: blog
+by: Ólafur Páll Geirsson
+title: "Refactor with scalafix v0.3"
+---
+
+I am happy to announce the release of [scalafix v0.3][scalafix], a library and tool to rewrite Scala source code.
+The long-term mission of scalafix is to help automate the migration of between Scala versions.
+However, as I hope to demonstrate in this post, scalafix can be used for more than just migrating between Scala versions.
+Scalafix can also be used for ad-hoc library and application migrations.
+
+Scalafix v0.3 uses the new scala.meta semantic API to provide a re-designed `Rewrite` and `Patch` API to implement custom refactorings.
+Let me explain what that means word by word.
+
+> Note. This is the second post on scalafix and scala.meta. You might be
+> interested in reading the first post
+> [here](http://scala-lang.org/blog/2016/10/24/scalafix.html).
+
+## Scala.meta semantic API
+
+Scala.meta recently announced the [first release of its semantic API][meta-1.6].
+This release is the product of close collaboration for the past several months between
+[@xeno_by][] at Twitter and [myself][@olafurpg] at the Scala Center.
+The objective of the semantic API is to provide operations to query information from the compiler.
+
+The first version of the scala.meta semantic API makes it possible to query for the resolved "symbol" of a name that appears in a Scala source file.
+A name is a reference to some definition, for example `println` or `scala.Predef.println`.
+A symbol is a unique identifier of a single definition.
+For example, `println` from the standard library has the symbol `_root_.scala.Predef.println(Ljava/lang/Object;)V.`.
+The compiler is responsible for resolving names to symbols.
+
+Scalahost is a compiler plugin in the scala.meta project that extracts symbols from the compiler and maps them to scala.meta syntax trees.
+Scalahost emits the extracted symbols into a "semantic database".
+The semantic database can be persisted to files on disk and loaded for later analysis.
+Semantic databases from different compilation units, potentially produced by different
+versions of the Scala compiler, can be merged.
+This opens possibilities for large-scale code analysis.
+
+The introduction of the scala.meta semantic API is a game changer for scalafix.
+The ability to resolve names to symbols opens possibilities for many scalafix rewrites.
```

Comment:
Good idea, I moved the conclusion of the "rewrite" section to the top of the section to make the flow better.